core/src/main/kotlin/org/eln2/sim/thermal/ThermalSimulator.kt

```diff
@@ -0,0 +1,63 @@
+package org.eln2.sim.thermal
+import org.eln2.sim.IProcess
+
+/**
+ * main thermal simulation class
+ */
+class ThermalSimulator: IProcess {
+    var thermalNodes = mutableSetOf<ThermalNode>()
+    var thermalEdges = mutableSetOf<ThermalEdge>()
+    override fun process(dt: Double) {
+        val threadList = mutableSetOf<Thread>()
+        thermalEdges.forEach {
+            val t = Thread()
+            t.run {
+                it.queueMoveEnergy(dt);
+            }
+            t.start()
+            threadList.add(t)
+        }
+        threadList.forEach { it.join() }
+        threadList.clear()
+        thermalNodes.forEach {
+            val t = Thread()
+            t.run {
+                it.commitThermalChanges();
+            }
+            t.start()
+            threadList.add(t)
+        }
+        threadList.forEach { it.join() }
+        thermalEdges.forEach {
+            it.clearRunBit()
+        }
+    }
+
+    fun connectNodes(n1: ThermalNode, n2: ThermalNode, dir: Int, distance: Double) {
```

Comment:
What is the purpose of `dir`? Perhaps write some javadoc since this looks like an API